pkg/v1/providers/ytt/02_addons/secretgen-controller/add_secretgen-controller.yaml

```diff
@@ -0,0 +1,24 @@
+#@ load("@ytt:overlay", "overlay")
+#@ load("@ytt:template", "template")
+#@ load("@ytt:data", "data")
+#@ load("@ytt:yaml", "yaml")
+#@ load("/lib/helpers.star", "ValuesFormatStr")
+
+#@ if data.values.PROVIDER_TYPE != "tkg-service-vsphere" and data.values.SECRETGEN_CONTROLLER_ENABLE:
+
+apiVersion: v1
+kind: Secret
+metadata:
+  name: #@ "{}-secretgen-controller-addon".format(data.values.CLUSTER_NAME)
+  namespace: #@ data.values.NAMESPACE
+  labels:
+    tkg.tanzu.vmware.com/addon-name: "secretgen-controller"
+    tkg.tanzu.vmware.com/cluster-name: #@ data.values.CLUSTER_NAME
+    clusterctl.cluster.x-k8s.io/move: ""
+  annotations:
+    tkg.tanzu.vmware.com/addon-type: "addons-management/secretgen-controller"
+type: tkg.tanzu.vmware.com/addon
+stringData:
+  values.yaml: #@ ValuesFormatStr.format(yaml.encode(""))
```

Comment:
I understand that we don't have any configuration values for secret-gen controller now. Please verify that encoding an empty string would work before merging. This is because we have never done this before, but theoretically an empty string would work.

Reply:
I verified by creating a mgmt cluster and workload cluster.
Secret generated is as follows:
```apiVersion: v1
kind: Secret
metadata:
  annotations:
    tkg.tanzu.vmware.com/addon-type: addons-management/secretgen-controller
  labels:
    clusterctl.cluster.x-k8s.io/move: ""
    tkg.tanzu.vmware.com/addon-name: secretgen-controller
    tkg.tanzu.vmware.com/cluster-name: mc
  name: mc-secretgen-controller-addon
  namespace: default
stringData:
  values.yaml: |
    #@data/values
    #@overlay/match-child-defaults missing_ok=True
    ---
    ""
type: tkg.tanzu.vmware.com/addon```

Do I need to verify any other thing @blc1996

Reply:
If you see the packageInstall of secretGen controller is successfully reconciled, we should be all good.